pkgs/build-support/go/README.md

```diff
@@ -0,0 +1,44 @@
+## Go toolchain/builder upgrade policy
+
+Go promises that "programs written to the Go 1 specification will continue to compile and run correctly, unchanged, over the lifetime of that specification" [1].
+Newer toolchain versions should build projects developed against older toolchains without problems.
+
+There are however Go packages depending on internal APIs of the toolchain/runtime/stdlib that are not covered by the Go compatibility promise.
+These packages may break on toolchain updates.
+We name packages that (often) break on toolchain updates `toolchain-breaking`.
+
+There is another set of packages that depends on the toolchain, but in another way:
+Packages providing development support for the Go language (like `gopls`, `golangci-lint`,...) must be compiled with the version they should be used for.
+If `gopls` is compiled for Go 1.23, it won't work for projects that require Go 1.24.
+We name packages that must be built with the latest toolchain to work as expected `toolchain-latest`.
```

Comment:
1. First I'd like to express support and say I think it's a good call to merge the current version. Details can be amended over time
2. The docs need to be comprehensible for people just starting out with Golang infra in Nixpkgs. As someone who knows more or less nil about the subject, I can't tell if `toolchain-latest` is to be a part of some path, or to be prepended to `pname`, or to attribute name, or why it's not part of `passthru`/`meta`. Maybe some examples would help

Reply:
Nit: designating the dependent clause with commas might make this easier to read

Reply:
Oh, so it's like "Definition(a toolchain-latest package) ...is a package that can only be built against the latest toolchain"?

Reply:
Maybe it's the backquotes that made me see it as a piece of code